queries/janet_simple/highlights.scm

```diff
@@ -0,0 +1,341 @@
+;; >> Explanation
+;; Parsers for lisps are a bit weird in that they just return the raw forms.
+;; This means we have to do a bit of extra work in the queries to get things
+;; highlighted as they should be.
+;;
+;; For the most part this means that some things have to be assigned multiple
+;; groups.
+;; By doing this we can add a basic capture and then later refine it with more
+;; specialized captures.
+;; This can mean that sometimes things are highlighted weirdly because they
+;; have multiple highlight groups applied to them.
+
+
+;; >> Literals
+
+(kwd_lit) @symbol
+(str_lit) @string @spell
+(long_str_lit) @string @spell
+(buf_lit) @string @spell
+(long_buf_lit) @string @spell
+(num_lit) @number
+(bool_lit) @boolean
+(nil_lit) @constant.builtin
+(comment) @comment @spell
+
+["'" "~"] @string.escape
+
+["|" "," ";"] @punctuation.special
```

Comment:
I don't really know what it should be, but for reference, I was going off of this bit in Clojure's file: https://github.com/nvim-treesitter/nvim-treesitter/blob/87faf372eb517077cc600c4e653e4fb3252d02ac/queries/clojure/highlights.scm#L31

Assuming these are comparable, should Clojure's change too then?

Reply:
hmm, are those used for the same thing? and if so, are they separators? I'll leave this up to your judgement

Reply:
These are all things that nearly always appear before other things.  To be more specific...

Both Clojure and Janet use `'` for "quoting".  The single quote goes in front of some other thing (e.g. `'(+ 1 1)` or '3) and is sometimes described as "preventing evaluation".

Clojure uses \` (backtick / backquote) for a specialized kind of quoting where one can selectively "unquote" some interior thing using `~` (e.g. \`(+ 1 ~a) could yield a list of 3 elements where the value for `a` gets "filled in" with whatever value it might have been bound to -- in contrast, in '(+ 1 a), `a` doesn't get evaluated / "filled in".

Janet uses `~` for this kind of specialized quoting and `,` for "unquoting".

Roughly, Clojure uses `#` in front of lists to indicate an anonymous function, whereas Janet uses `|`.  Examples are: In Clojure, `#(+ % 2)` - anonymous function that increments the argument by 2.  In Janet, `|(+ $ 2)` - similar idea (though in Janet what comes after doesn't have to be something in parens).

Clojure uses `~@` for "unquoting + splicing" where are Janet uses `;` for splicing (like the JS' spread syntax).

I wouldn't call them "separators", but I'm not sure what you meant by that.

Is this perhaps a case of there not necessarily being a good place to put them?